packages/renderer/src/lib/preferences/LineWithLinks.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+export let lineText: string;
+
+interface Segment {
+  text: string;
+  href?: string;
+}
+
+function createLink(text: string): Segment {
+  const match = text.match(/\[(?<text>.*)\]\[(?<url>.*)\]/);
+  return {
+    text: match.groups.text,
+    href: match.groups.url,
+  };
+}
+
+let text = lineText;
+let segments: Segment[] = [];
+while (text.length > 0) {
+  const match = text.search(/\[.*\]\[.*\]/);
+  if (match > 0) {
+    // it is a link
+    segments.push({ text: text.substring(0, match) });
+    const linkMatch = text.match(/\[(?<text>.*)\]\[(?<url>.*)\]/);
```

Comment:
you could call the `createLink` function as it does the same

Reply:
Ops, just forgot to call it.